internal/keepers/service.go

```diff
@@ -202,17 +183,57 @@ func (s *onDemandUpkeepService) parallelCheck(ctx context.Context, keys []types.
 	wg.Wait()
 	close(done)
 
+	if wResults.Total() == 0 {
+		s.logger.Printf("no network calls were made for this sampling set")
+	} else {
+		s.logger.Printf("worker call success rate: %.2f; failure rate: %.2f; total calls %d", wResults.SuccessRate(), wResults.FailureRate(), wResults.Total())
+	}
+
 	s.logger.Printf("sampling cache hit ratio %d/%d", cacheHits, len(keys))
 
+	// multiple network calls can result in an error while some can be successful
+	// in the case that all workers encounter an error, bubble this up as a hard
+	// failure of the process.
+	if wResults.Total() > 0 && wResults.Total() == wResults.Failure && wResults.LastErr != nil {
+		return samples.Values(), fmt.Errorf("%w: too many errors in parallel worker process; last error provided", wResults.LastErr)
```

Comment:
At this point I see this case as a hard failure. Even if multiple upkeeps were pulled from cache, if we see all failures coming from the RPC it should be considered a faulty node.

Reply:
not sure, imo if we have some results then better to pass it. If RPC is indeed down then later stages would give the error. Don't think it will have any practical effect though so happy to have it as is

Reply:
The cached results are included with the error here anyway so it is up to the calling function what to do with that error.